src/vstest.console/CommandLine/InferHelper.cs

```diff
@@ -133,35 +133,24 @@ public Architecture AutoDetectArchitecture(IList<string?>? sources, Architecture
     public Framework AutoDetectFramework(IList<string?>? sources, out IDictionary<string, Framework> sourceToFrameworkMap)
     {
         sourceToFrameworkMap = new Dictionary<string, Framework>();
-        Framework framework = Framework.DefaultFramework;
 
         if (sources == null || sources.Count == 0)
```

Comment:
I am not filtering `null` sources here because the simplest solution is to actually return `Framework` instead of `FrameworkName` on `DetermineFramework`. This way I can solve the "problem" of instances equality.